qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -464,6 +465,67 @@ def to_list(
             ]
         return self._to_dense_label_data()
 
+    def to_matrix(self, sparse: Optional[bool] = True) -> Union[csc_matrix, np.ndarray]:
+        """Convert to a matrix representation over the full fermionic Fock space in occupation number
+        basis. The basis states are ordered in increasing bitstring order as 0000, 0001, ..., 1111.
+
+        Args:
+            sparse: If true, the matrix is returned as a sparse csc_matrix, else it is returned as a
+            dense numpy array.
+
+        Returns:
+            The matrix of the operator in the Fock basis (scipy.sparse.csc_matrix or numpy.ndarray
+            with dtype=numpy.complex128)
+        """
+
+        csc_data, csc_col, csc_row = [], [], []
+
+        # loop over all columns of the matrix
+        for col_idx in range(2 ** self.register_length):
+            initial_occupations = [
+                occ == "1" for occ in np.binary_repr(col_idx, self.register_length)
```

Comment:
```suggestion
                occ == "1" for occ in f"{col_idx:0{self.register_length}b}"
```
This performs slightly better than the numpy-call